tmt/steps/execute/internal.py

```diff
@@ -304,6 +237,9 @@ def _handle_reboot(self, test, guest):
                     "Guest does not support soft reboot, "
                     "trying hard reboot.")
                 guest.reboot(hard=True)
+            # Prepare scripts after the reboot for podman
+            if isinstance(guest, GuestContainer):
+                self.prepare_scripts(guest)
```

Comment:
This is a questionable approach because it requires external knowledge of `GuestContainer` implementation. Basically users of various `Guest` classes must be aware of their internals, implementation, and side effects, and that's breaking the encapsulation. Users of these classes should not be forced into this.

But, there are, obviously, corner cases where guest implementation has side effects the main code does not expect, and in that case, I for one would extend the `Guest` API. In this case, I would add a boolean flag to the `Guest` class, e.g. `reboot_resets_script` (default `False`), which would become part of the "public" API of the `Guest` class, and any outside code could rely on it instead of knowing which classes need special treatment. The flag would be under full control of each `Guest` subclass, therefore the fact a particular guest implementation needs special treatment lies entirely in the hands of its developers.

Reply:
Actually I believe this one can go away. I will check, one less crap.

Reply:
it can be just removed, the problem I got burned on was enumerate :D added comment for future folks wondering why enumerate cannot be used :dagger: